src/physics/base/ModelIdGenerator.hh

```diff
@@ -0,0 +1,29 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ModelIdGenerator.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class for constructing model IDs.
+ */
+class ModelIdGenerator
```

Comment:
Having sequential model ids created at initialization seems pretty good, in the sense that we don't have worry when adding a new model to the code. Still, we can't know for sure which ID represents which model (since we cannot have a 'get_model_name()` kind of function in the `Model` class), unless we personally check the model construction ordering in a specific application. Do you think we might miss having a clear model id (such as an enum) for debug/verbose runs?

Reply:
The `PhysicsParams` class (not committed in this MR) has an accessor `const Model& model(ModelId) const;` for accessing the model. But you're right that a label would make things clearer for debugging -- I'll add that.